### PR TITLE
Fix NPE in OCFileListFragment when OCFile is null

### PR DESCRIPTION
**Pull request by ezaquarii**
1. Fixed NPE
2. Audited all use of nullable field in all called APIs
3. Updated called APIs nullability documentation

Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

- [x] No test as this is a speculative fix for a tricky edge case + audit changes

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 1]
```diff
@@ -1330,10 +1330,10 @@ private void setGridSwitchButton() {
      * Determines if user set folder to grid or list view. If folder is not set itself,
      * it finds a parent that is set (at least root is set).
      *
-     * @param folder Folder to check.
+     * @param folder Folder to check or null for root folder
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java:1333]: Is this really true?
Root folder should also just be a OCFile? 🤔

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java:1333]: Created issue #6388
